packages/toolpad-app/src/components/EditableTreeItem.tsx

```diff
@@ -113,6 +112,8 @@ export default function EditableTreeItem({
     (event: React.KeyboardEvent<HTMLInputElement>) => {
       if (event.key === 'Enter') {
         handleConfirm();
+      } else if (event.key === 'Backspace') {
+        event.stopPropagation();
```

Comment:
Oh, also maybe we should just stop propagation at the start no matter what they key is, to avoid future issues?